cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -797,6 +797,21 @@ public NamespaceMeta apply(AppMds input) throws Exception {
   }
 
   @Override
+  public void updateNamespace(final NamespaceMeta metadata) {
+    Preconditions.checkArgument(metadata != null, "Namespace metadata cannot be null.");
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
```

Comment:
Should we start throwing checked exceptions from the ``Store``? Or keep it consistent right now and fix it all later?